Source/Core/Core/HW/GCMemcard/GCMemcardDirectory.cpp

```diff
@@ -703,12 +703,12 @@ void MigrateFromMemcardFile(const std::string& directory_name, int card_index)
                                                 Config::Get(Config::MAIN_MEMCARD_B_PATH);
   if (File::Exists(ini_memcard))
   {
-    GCMemcard memcard(ini_memcard.c_str());
-    if (memcard.IsValid())
+    auto memcard = GCMemcard::Open(ini_memcard.c_str());
```

Comment:
How about using structured bindings instead of `.first`/`.second`?
```suggestion
    auto [error_code, memcard] = GCMemcard::Open(ini_memcard.c_str());
```